Revamp of user role doc for crate su info (#12)
crate/cloud-reference: docs/user-roles.rst

    organization.

A *project member* has read-only access to the project.

.. _user-ops:

User operations
===============

.. _system-user:

``System`` user
---------------

In CrateDB Cloud, the backend uses a user called ``system`` in order to perform
Cloud cluster upgrades, backups and scaling functions, among other things.

> .. WARNING::
>     The user ``system`` is essential for CrateDB Cloud to function as intended.

Reviewer: ```suggestion
.. WARNING::

    The user ``system`` is essential for CrateDB Cloud to function as intended.
```

    While it is not normally accessible through the CrateDB Console or the
    Croud CLI, it can be accessed through the CrateDB Admin UI or any other SQL
    client. It is important not to edit or delete this user in any way.
    Otherwise, the functioning of Cloud clusters may be compromised.

.. _crate-user:

``Crate`` user
--------------

In CrateDB, when you create a cluster node (through whatever method), a
``crate`` user is automatically generated. ``crate`` is a superuser and can
perform all possible operations. For CrateDB Cloud purposes, however, this user
is not exposed to the Cloud customer, who instead creates their own users
according to the user roles and privileges described above. (For information on

> how to do so, see our documentation on `adding users`_).

Reviewer 2: I think you mix up two different user types here: On the one hand there are Cloud users, and on the other hand there are CrateDB users.

The former are the ones you log in with to the console and has privileges such as org member/admin, project member/admin.

The latter are `crate`, `system` and the one that is specified upon cluster creation. Both the `crate` and `system` user are visible to the user generated user, but cannot be used. The user generated user has [AL privileges](https://crate.io/docs/crate/reference/en/latest/admin/privileges.html#al).

https://crate.io/docs/crate/reference/en/latest/admin/user-management.html
> When CrateDB is started, the cluster contains one predefined superuser. This user is called crate. It is not possible to create any other superusers.

> how to do so, see our documentation on `adding users`_).

Reviewer 3: I think there's a misunderstanding:
a "CrateDB Cloud" user is different from a "CrateDB" user.
The former is used to interact with the CreatDB Cloud application. That involves creating org/projects, adding/removing other users, deploying clusters, etc..
The "CrateDB" user is merely a user of the database service that is offered on cratedb cloud.

So these two types of users are not connected at all.
This should only be about "CrateDB" users.
The CrateDB Cloud user creates their own "CrateDB" user when they deploy a new cluster (that's what username and password is for on step 2 of the wizard). And it is about this very database user that doesn't have access to the operations below. It has nothing to do with the CrateDB Cloud user.

So when a new cluster is being deployed, the CrateDB instance creates the "Crate" user (automatically from the deployment script) and a second database user (from the information from the wizard) so the customer can access CrateDB with the latter.

Hope that makes things clearer. :)

    organization.

A *project member* has read-only access to the project.

.. _user-ops:

> User operations

Reviewer 2: What about naming this section "Database users" or "CrateDB users" (vs. Cloud users)?

===============

> This section describes issues relating to user operations that derive from the

Reviewer 2: I don't think it is very positive to talk about "issues" in the first sentence. This paragraph could state that there are different types of database users and that their differences are described in more detail below.

Author: That's a great idea! It changes the nature of the doc a bit but I think for the better.

architecture of CrateDB and CrateDB Cloud. They are here appended to ensure the
CrateDB Cloud customer is aware of the constraints they impose.

.. _system-user:

``System`` user
---------------

In CrateDB Cloud, the backend uses a user called ``system`` in order to perform
Cloud cluster upgrades, backups and scaling functions, among other things.

.. WARNING::

    The user ``system`` is essential for CrateDB Cloud to function as intended.

>     While it is not normally accessible through the CrateDB Console or the

Reviewer 3: CrateDB Cloud Console

    Croud CLI, it can be accessed through the CrateDB Admin UI or any other SQL
    client. It is important not to edit or delete this user in any way.
    Otherwise, the functioning of Cloud clusters may be compromised.

.. _crate-user:

``Crate`` user
--------------

In CrateDB, when you create a cluster node (through whatever method), a
``crate`` user is automatically generated. ``crate`` is a superuser and can

> perform all possible operations. Next to the ``crate`` user there is the
> regular database user, created as part of the CrateDB Cloud cluster
> deployment wizard when deploying a cluster through `AWS`_ or `Azure`_.
> 
> Because ``crate`` is a superuser, there are certain operations that the *crate*
> user can perform which are not possible for the regular database user. As of
> CrateDB 4.2.1, the list of such operations is as follows:

Reviewer 2: I think we can also distinguish hierarchically between the crate user and regular users., especially because otherwise it looks very much as if the database user would also be a superuser.

```suggestion
perform all possible operations.

.. _regular-user:

Regular user
------------

Next to the ``crate`` user there is the
regular database user, created as part of the CrateDB Cloud cluster
deployment wizard when deploying a cluster through `AWS`_ or `Azure`_.

Because the regular user is as a database user with `AL`_ privileges, there are certain operations that this user cannot do. As of
CrateDB 4.2.1, the list of such operations is as follows:
```

Reviewer 2: In general, the distinction between Cloud users and CrateDB users is much clearer now.
